File: Assignments/Intro to Comp Sci/Python/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

words = 0
sentences = 0
letters = 0

for i in range(len(text)):
    if i == 0:
        if text[i].isalpha():
            words += 1
    elif text[i].isalpha() and (text[i-1] == " " or text[i-1] == '"'):
        words +=1
    if text[i].isalpha():
        letters += 1
    if text[i] in ["!","?","."]:
        sentences += 1


L = (letters/words)*100;
S = (sentences/words)*100;
index = 0.0588 * L - 0.296 * S - 15.8
grade = round(.0588*L-.296*S-15.8)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade " + str(grade))

File: Assignments/Intro to Comp Sci/C/readability/readability.c
//include standard and cs50 and letter classification and math functions
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
//define main function
int main(void)
{
    //prompt user for text and save as string
    string text=get_string("Text: ");
    //count letters in text
    int letters=0;
    for (int i=0; text[i]!='\0'; i++)
    {
        //only count if alphabet letter
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    //after iterating through text, pring number of letters counted
    printf("%i letter(s)\n",letters);
    //iterate through text to count words
    int words=0;
    for (int j=0;text[j]!='\0';j++)
    {
        //check if we're on the first character, special rules
        if (j==0)
        {
            //if it's a character, increment counter. if not, do nothing
            if isalpha(text[j])
            {
                words++;
            }
        }
        //check if current character is an alpha and if previous character is a space or "
        if (isalpha(text[j]) && (isblank(text[j-1]) || text[j-1]=='"'))
        {
            //increment word counter
            words++;
        }
    }
    //display number of words counted
    printf("%i word(s)\n", words);
    //count sentences by counting periods, question marks, and exclamation points
    int sentences=0;
    for (int k=0;text[k]!='\0';k++)
    {
        //check if char is period ,exclamation, or question
        if (text[k]=='.'||text[k]==(int)'!'||text[k]==(int)'?')
        {
            //increment counter
            sentences++;
        }
    }
    //print sentences counted
    printf("%i sentence(s)\n",sentences);
    //process formula to calculate grade level
    float L = ((float)letters/words)*100;
    float S = ((float)sentences/words)*100;
    float grade = .0588*L-.296*S-15.8;
    //print float grade
    printf("%f\n",grade);
    //check if grade is above 16
    if (grade>=16)
    {
        //print 16+
        printf("Grade 16+");
    }
    //check if grade is less than 1
    else if (grade<1)
    {
        //print before grade 1
        printf("Before Grade 1");
    }
    //output all other grade values
    else
    {
        printf("Grade %i",(int)round(grade));
    }
    //finish with a new line for next input
    printf("\n");
}